### DISCUSSION TITLE
'zfs send' failed due to filesystem error, but no details in zrepl log

### DISCUSSION THREAD

**Discussion by kpfleming** [zrepl/zrepl#633] [General]
This weekend I setup a new system and when I attempted to use zrepl to replicate the filesystem for the first time, it failed with a fairly generic 'command failed with error' message in both logs (client and server).

When I tried the same `zfs send` command by hand, I learned that it was failing due to checksum errors in the source filesystem. After removing the damaged files and creating a new snapshot, replication was successful.

Is there any way the actual error messages emitted by `zfs send` or `zfs receive` could be included in the zrepl log? I tried with the log level set to `debug` but still didn't see anything which indicated the source of the problem.

**Comment by problame** [comment]
Yes, we should provide those detailed errors. This is more of an issue than a discussion. Created https://github.com/zrepl/zrepl/issues/635 to track it.
